app/controllers/application_controller.rb

```diff
@@ -120,7 +120,8 @@ def set_locale
       end
     end
     I18n.locale = locale
```

Comment:
This function is quite messy. Is there a clearer way to write it?

If it has to be that messy, please add a comment explaining what it does as that is currently not obvious.